cache.go

```diff
@@ -1,91 +1,266 @@
+// LRU functionality largely inspired by https://github.com/golang/groupcache
 package agecache
 
 import (
+	"container/list"
+	"errors"
 	"sync"
 	"sync/atomic"
 	"time"
-
-	"github.com/hashicorp/golang-lru/simplelru"
 )
 
 // Stats hold cache statistics.
 type Stats struct {
-	Hits   int64 // number of cache hits.
-	Misses int64 // number of cache misses
+	Count     int64 // Gauge, number of items in the cache
```

Comment:
Should we add gauge for the `size` too? saves us from having to look at configs :P